rcljava/src/main/java/org/ros2/rcljava/qos/QoSProfile.java

```diff
@@ -44,39 +59,124 @@ public final History getHistory() {
     return this.history;
   }
 
+  public final QoSProfile setHistory(final History history) {
+    this.history = history;
+    return this;
+  }
+
   public final int getDepth() {
     return this.depth;
   }
 
+  public final QoSProfile setDepth(final int depth) {
+    this.depth = depth;
+    return this;
+  }
+
   public final Reliability getReliability() {
     return this.reliability;
   }
 
+  public final QoSProfile setReliability(final Reliability reliability) {
+    this.reliability = reliability;
+    return this;
+  }
+
   public final Durability getDurability() {
     return this.durability;
   }
 
+  public final QoSProfile setDurability(final Durability durability) {
+    this.durability = durability;
+    return this;
+  }
+
+  public final Duration getDeadline() {
+    return this.deadline;
+  }
+
+  public final QoSProfile setDeadline(final Duration deadline) {
+    this.deadline = deadline;
+    return this;
+  }
+
+  public final Duration getLifespan() {
+    return this.lifespan;
+  }
+
+  public final QoSProfile setLifespan(final Duration lifespan) {
+    this.lifespan = lifespan;
+    return this;
+  }
+
+  public final Liveliness getLiveliness() {
+    return this.liveliness;
+  }
+
+  public final QoSProfile setLiveliness(final Liveliness liveliness) {
+    this.liveliness = liveliness;
+    return this;
+  }
+
+  public final Duration getLivelinessLeaseDuration() {
+    return this.livelinessLeaseDuration;
+  }
+
+  public final QoSProfile setLivelinessLeaseDuration(final Duration livelinessLeaseDuration) {
+    this.livelinessLeaseDuration = livelinessLeaseDuration;
+    return this;
+  }
+
   public final boolean getAvoidROSNamespaceConventions() {
     return this.avoidROSNamespaceConventions;
   }
 
+  public final QoSProfile setAvoidROSNamespaceConventions(final boolean avoidROSConventions) {
+    this.avoidROSNamespaceConventions = avoidROSConventions;
+    return this;
+  }
+
+  public static final QoSProfile defaultProfile() {
+    return new QoSProfile(10);
+  }
+
+  public static final QoSProfile systemDefault() {
+    return new QoSProfile(
+      History.SYSTEM_DEFAULT, QoSProfile.DEPTH_SYSTEM_DEFAULT,
+      Reliability.SYSTEM_DEFAULT, Durability.SYSTEM_DEFAULT, false);
+  }
+
+  public static final QoSProfile sensorData() {
+    return new QoSProfile(
+      History.KEEP_LAST, 5, Reliability.BEST_EFFORT, Durability.VOLATILE, false);
```

Comment:
What do you think about creating the Java objects from the `rcl` definitions? Then we don't go out of sync if the `rcl` definitions ever change (though I don't know how likely that is to happen). Otherwise, these values appear arbitrary.

Reply:
I agree, though that would need a refactor unrelated to this PR.

Reply:
Please add a TODO at least so we are reminded.

Reply:
Done in https://github.com/osrf/ros2_java/pull/20/commits/138bed43e47a9c6cebdf64f303ed22ca82287ab0